client/dashboard/domains/domain-security/dnssec-record-textarea.tsx

```diff
@@ -7,37 +7,44 @@ interface DnsSecRecordTextareaProps {
 	label: string;
 }
 
+const MIN_ROWS = 2;
+
 export function DnsSecRecordTextarea( { value, label }: DnsSecRecordTextareaProps ) {
 	const textareaRef = useRef< HTMLTextAreaElement >( null );
-	const [ textareaHeight, setTextareaHeight ] = useState< number >( 0 );
-
-	// Calculate and set textarea height
-	const updateTextareaHeight = () => {
-		if ( textareaRef.current ) {
-			const textarea = textareaRef.current;
-			// Reset height to auto to get the correct scrollHeight
-			textarea.style.height = 'auto';
-			const newHeight = textarea.scrollHeight;
-			setTextareaHeight( newHeight );
+	const [ textareaRows, setTextareaRows ] = useState< number >( MIN_ROWS );
+
+	// Calculate and set textarea rows
+	const updateTextareaRows = () => {
+		const textareaElement = textareaRef.current;
+		if ( ! textareaElement ) {
+			return;
 		}
+
+		// Reset to minimum rows first
+		textareaElement.rows = MIN_ROWS;
```

Comment:
You're setting `textareaElement.rows` directly and then also calling `setTextareaRows`, which could cause unnecessary re-renders.